contracts/src/v0.8/automation/dev/test/AutomationRegistry2_3.t.sol

```diff
@@ -443,3 +504,133 @@ contract SetConfig is SetUp {
     return bytes32((prefix & prefixMask) | (h & ~prefixMask));
   }
 }
+
+contract NOPsSettlement is SetUp {
+  event NOPsSettledOffchain(address[] payees, uint256[] balances);
+
+  function testSettleNOPsOffchainRevertDueToUnauthorizedCaller() public {
+    (IAutomationRegistryMaster2_3 registry, ) = deployAndConfigureAll(AutoBase.PayoutMode.ON_CHAIN);
+
+    vm.expectRevert(abi.encodeWithSelector(IAutomationRegistryMaster2_3.OnlyFinanceAdmin.selector));
+    registry.settleNOPsOffchain();
+  }
+
+  function testSettleNOPsOffchainRevertDueToOffchainSettlementDisabled() public {
+    (IAutomationRegistryMaster2_3 registry, ) = deployAndConfigureAll(AutoBase.PayoutMode.OFF_CHAIN);
```

Comment:
I love how fast / easy it is to deply and configure the registry in this test framework. we could **never** do this in hardhat.